core/src/messages/agent.rs

```diff
@@ -105,14 +105,15 @@ impl BackendStatsMessage {
 impl BackendStatsMessage {
     pub fn from_stats_message(
         backend_id: &BackendId,
-        stats_message: &Stats,
-    ) -> Option<BackendStatsMessage> {
+        prev_stats_message: &Stats,
+        cur_stats_message: &Stats,
+    ) -> BackendStatsMessage {
         // based on docs here: https://docs.docker.com/engine/api/v1.41/#tag/Container/operation/ContainerStats
 
         //memory
-        let mem_naive_usage = stats_message.memory_stats.usage.unwrap_or_default();
-        let mem_available = stats_message.memory_stats.limit.unwrap_or(u64::MAX);
-        let mem_stats = stats_message.memory_stats.stats;
+        let mem_naive_usage = cur_stats_message.memory_stats.usage.unwrap_or_default();
+        let mem_available = cur_stats_message.memory_stats.limit.unwrap_or(u64::MAX);
```

Comment:
Rather than using default values (and returning an incorrect result), if these are `None`, we should return an error.

Usually I'd say let's make that a follow-up PR since the problem pre-existed this PR, but since this PR changes the return type of `from_stats_message`, I suggest the following as part of this PR:

- rename `from_stats_message` to `from_stats_messages`
- change the return type to `Result<BackendStatsMessage>`
- if any field is missing, return an anyhow error stating which field is missing